### PR TITLE
gettext 0.22.5

### PR DESCRIPTION
**Pull request by xhochy**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xhochy** [comment]
@conda-forge/gettext @conda-forge/core This is ready for review. It is a major rework of this package and thus I would appreciate a good review. I have also set the target_channel to `gettext_dev` to build glib and similar libs first before pushing it to main as there have been problems before and I want to be extra sure to not break conda-forge with this.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -39,23 +36,350 @@ requirements:
...
+    test:
+      commands:
+        - test -f ${PREFIX}/lib/libgettextpo$SHLIB_EXT               # [not win]
+        - if not exist %PREFIX%\Library\lib\gettextpo.dll.lib exit /b 1    # [win]
+    about:
+      home: https://www.gnu.org/software/gettext/
+      license: GPL-3.0-or-later
+      license_file: COPYING
+      summary: library for interfacing programmatically with GNU gettext .po files
+  - name: gettext-bin
```
Comment (isuruf) [inline recipe/meta.yaml:283]:
```suggestion
  - name: gettext-tools
```

File: recipe/install-asprintf-devel.bat [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+if errorlevel 1 exit 1
+
+if not exist %LIBRARY_PREFIX%\lib md %LIBRARY_PREFIX%\lib
+if errorlevel 1 exit 1
+
+copy gettext-runtime\libasprintf\.libs\asprintf.dll.lib %LIBRARY_PREFIX%\lib\asprintf.dll.lib
+if errorlevel 1 exit 1
+
+@rem Enforce dynamic linkage
+copy gettext-runtime\libasprintf\.libs\asprintf.dll.lib %LIBRARY_PREFIX%\lib\asprintf.lib
```
Comment (pkgw) [inline recipe/install-asprintf-devel.bat:16]: Just to double-check, it's intentional here to copy the `.dll.lib` to the `.lib` path? (Here and for the other -devel packages)

File: recipe/install-libintl-devel.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+set -euxo pipefail
+
+mkdir -p ${PREFIX}/include
+cp ./gettext-runtime/intl/libintl.h ${PREFIX}/include/libintl.h
+
+pushd ${PREFIX}/lib
+if [[ "${target_platform}" == osx-* ]]; then
+  ln -s libintl.*.dylib libintl.dylib
+else
+  ln -s libintl.so.* libintl.so
```
Comment (pkgw) [inline recipe/install-libintl-devel.sh:12]: Other scripts hardcode checking for the particular library version (e.g., `.so.0`) — should this script do likewise?

File: recipe/meta.yaml [hunk 4]
```diff
@@ -39,23 +36,350 @@ requirements:
...
+        - {{ native }}binutils                 # [win]
+        - {{ native }}crt-git                  # [win]
+        - {{ compiler('c') }}                # [not win]
+        - {{ compiler('cxx') }}              # [not win]
+        - make                               # [unix]
+      host:
+        - libiconv  # [not linux]
+    test:
+      commands:
+        # Only the actual binary (versioned) version should be in this package
```
Comment (pkgw) [inline recipe/meta.yaml:75]: Typo "version" for "variant" here?

File: recipe/meta.yaml [hunk 5]
```diff
@@ -39,23 +36,350 @@ requirements:
...
+        - make                               # [unix]
+      host:
+        - libiconv  # [not linux]
+        - {{ pin_subpackage('libintl', exact=true) }}
+      run:
+        - {{ pin_subpackage('libintl', exact=true) }}
+    test:
+      commands:
+        - test -f ${PREFIX}/lib/libintl${SHLIB_EXT}                   # [osx]
+        - if not exist %PREFIX%\Library\lib\intl.dll.lib exit /b 1    # [win]
```
Comment (pkgw) [inline recipe/meta.yaml:119]: Also test for `intl.lib` here?

File: recipe/meta.yaml [hunk 6]
```diff
@@ -39,23 +36,350 @@ requirements:
...
+        - {{ native }}binutils                 # [win]
+        - {{ native }}crt-git                  # [win]
+        - {{ compiler('c') }}                # [not win]
+        - {{ compiler('cxx') }}              # [not win]
+        - make                               # [unix]
+      host:
+        - libiconv  # [not linux]
+    test:
+      commands:
+        # Only the actual binary (versioned) version should be in this package
```
Comment (pkgw) [inline recipe/meta.yaml:152]: Same query as above.

File: recipe/meta.yaml [hunk 7]
```diff
@@ -39,23 +36,350 @@ requirements:
...
+        - test -f ${PREFIX}/lib/libasprintf.so.0                      # [linux]
+        - if not exist %PREFIX%\Library\bin\asprintf-*.dll exit /b 1  # [win]
+        # The unversioned variants as well as the lib on Windows should be in the main gettext package.
+        - test ! -f ${PREFIX}/lib/libasprintf$SHLIB_EXT               # [not win]
+        - if exist %PREFIX%\Library\lib\asprintf.dll.lib exit /b 1    # [win]
+    about:
+      home: https://www.gnu.org/software/gettext/
+      license: LGPL-2.1-or-later
+      license_file: COPYING
+      summary: Decomposed printf argument list
```
Comment (pkgw) [inline recipe/meta.yaml:163]: This summary is a bit odd; one alternative that I can find is: "C-style formatted output in C++"

File: recipe/meta.yaml [hunk 8]
```diff
@@ -39,23 +36,350 @@ requirements:
...
+        - make                               # [unix]
+      host:
+        - libiconv  # [not linux]
+        - {{ pin_subpackage('libasprintf', exact=True) }}
+      run:
+        - {{ pin_subpackage('libasprintf', exact=True) }}
+    test:
+      commands:
+        - test -f ${PREFIX}/lib/libasprintf${SHLIB_EXT}               # [not win]
+        - if not exist %PREFIX%\Library\lib\asprintf.dll.lib exit /b 1    # [win]
```
Comment (pkgw) [inline recipe/meta.yaml:196]: Once again: also check for the plain `.lib`?

File: recipe/meta.yaml [hunk 9]
```diff
@@ -39,23 +36,350 @@ requirements:
...
+      commands:
+        - msgfmt.exe --help  # [win]
+        - msgfmt.exe -o an.gmo an.po                 # [win]
+        - msgfmt -o an.gmo an.po                     # [not win]
+    about:
+      home: https://www.gnu.org/software/gettext/
+      # The binaries of gettext are the GPL-3.0 licensed part.
+      license: GPL-3.0-or-later
+      license_file: COPYING
+      summary: Internationalization package
```
Comment (pkgw) [inline recipe/meta.yaml:322]: Something like "Internationalization package (executable tools)" here?

**Review by pkgw — COMMENTED** [review]
This looks really good! I haven't thought through every `pin_subpackage` but overall it all looks like this is making sense. I have some small-bore queries but no major concerns.

File: recipe/meta.yaml [hunk 10]
```diff
@@ -39,23 +36,350 @@ requirements:
...
+        - {{ posix }}patch                     # [win]
+        - {{ posix }}sed                       # [win]
+        - {{ native }}binutils                 # [win]
+        - {{ native }}crt-git                  # [win]
+        - {{ compiler('c') }}                # [not win]
+        - {{ compiler('cxx') }}              # [not win]
+        - make                               # [unix]
+      host:
+        - libiconv  # [not linux]
+        - {{ pin_subpackage('gettext-bin', exact=True) }}
```
Comment (xhochy) [inline recipe/meta.yaml:356]:
```suggestion
        - {{ pin_subpackage('gettext-tools', exact=True) }}
```

**Comment by ocefpaf** [comment]
Thanks @xhochy and @pkgw! BTW, don't wait for me to merge things like this. I will be "away" for a few weeks.

**Pull request merged**